src/services/client.rs

```diff
@@ -0,0 +1,143 @@
+use super::{Connect, Login, Disonnect};
+use crate::websocket::WsClient;
+use actix::prelude::*;
+use slog;
+use slog::{debug, info};
+use std::collections::HashMap;
+use uuid::Uuid;
+
+extern crate slog_term;
+
+// Types
+
+#[derive(Clone, Hash, PartialEq, Eq)]
+pub struct UserId(pub String);
+
+// Actor
+pub struct ClientActor {
+    logger: slog::Logger,
+    clients: HashMap<Addr<WsClient>, InternalClient>,
+}
+
+impl ClientActor {
+    pub fn new(logger: slog::Logger) -> Self {
+        ClientActor {
+            logger,
+            clients: HashMap::new(),
+        }
+    }
+}
+
+impl Default for ClientActor {
+    fn default() -> Self {
+        unimplemented!(
+            "Client actor can't be unitialized using default because it needs a logger"
+        )
+    }
+}
+
+impl Actor for ClientActor {
+    type Context = Context<Self>;
+
+    fn started(&mut self, _ctx: &mut Self::Context) {
+        info!(self.logger, "Client actor started");
+    }
+}
+
+impl Handler<Connect> for ClientActor {
+    type Result = ();
+
+    // reconnection problem: see if the client exists already? maybe if the client connects and sends Id we can connect them back together somehow?
+    // otherwise we'll create a new addr and a new uuid and the client will have to log in again.
+    // this is required as we'd like to be able to restart the web socket server for updates while people are connected.
+
+    fn handle(&mut self, msg: Connect, _ctx: &mut Context<Self>) -> Self::Result {
+        debug!(self.logger, "Adding new client to ClientActor");
+
+        let uuid = Uuid::new_v4().to_hyphenated().to_string();
+
+        debug!(self.logger, "Adding new client {}", uuid); // @todo: use real slog
+
+        let client = InternalClient{
+            username: "".to_owned(), // we don't know this at this point in time
+            id: uuid,
+        };
+
+        self.clients.insert(msg.addr.to_owned(), client.clone());
+
+        // Tell client its ID
+        msg.addr.do_send(IncomingNewClient(client.clone()));
```

Comment:
I'm guessing the compiler will optimize this bit anyway, but clone is not necessary here as we don't need client after this line.